### ISSUE TITLE
HHVM using all of /run tmpfs

### ISSUE THREAD

**Issue by collinbarrett** [facebook/hhvm#7240]
I apologize if this is the wrong place to submit this request. Please redirect me if so.

### HHVM Version
HipHop VM 3.14.3
Ubuntu 16.04.1

### Expected result
Less memory usage on /run tmpfs.

### Actual result
Reasonable memory usage on /run tmpfs.

### Details

I have been running HHVM on Ubuntu 14.04 to host several WordPress sites for over a year. It's been great. I just built a new 16.04 server, and I am having problems with HHVM eating up all of my /run tmpfs space. I found this issue when trying to restart NGINX. When doing so, I get...

> Job for nginx.service failed because a configured resource limit was exceeded. See "systemctl status nginx.service" and "journalctl -xe" for details.

I then have to first stop HHVM, then restart NGINX, then start HHVM to resolve. Is there a configuration change I need to make to resolve this? Do I need to resize my /run tmpfs (if so, how?)?

Thanks for any help you are able to provide.

### More Info:

Results of `df -h`

> Filesystem      Size  Used Avail Use% Mounted on
udev            238M     0  238M   0% /dev
tmpfs            49M   49M     0 100% /run
/dev/vda1        20G  6.7G   12G  36% /
tmpfs           245M     0  245M   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           245M     0  245M   0% /sys/fs/cgroup
tmpfs            32M  1.3M   31M   4% /mnt/ramdisk
tmpfs            49M     0   49M   0% /run/user/999
tmpfs            49M     0   49M   0% /run/user/1000

Results of `sudo du -h /run --max-depth=1 | sort -hr`

> 49M     /run/hhvm
49M     /run
356K    /run/udev
188K    /run/systemd
16K     /run/network
...

**Comment by Orvid** [comment]
What files are in `/run/hhvm`? I don't think we should be sticking anything nearly that big in there...

**Comment by collinbarrett** [comment]
It's all one big hhvm.hhbc file, it looks like. Quick Googling says it might be something WordPress-related. Reading about it now.

**Comment by collinbarrett** [comment]
Ok, so, in `~/` I see `-rw-r--r-- 1 myuser myuser 143003648 Jul 21 11:14 .hhvm.hhbc`.
And, in `/run/hhvm` I see `-rw-r--r-- 1 www-data www-data 50541568 Jul 21 12:36 /run/hhvm/hhvm.hhbc`

Must be a configuration issue... I've never done any configuration with HHVM before; just installed and ran. Any ideas welcome.

**Comment by collinbarrett** [comment]
Ok, I found `hhvm.repo.central.path = /var/run/hhvm/hhvm.hhbc` in my server.ini. That is what is generating the file. It seems like the hhvm.hhbc should be stored on disk rather than in RAM, though. I installed, what I thought, was a default HHVM installation (Step 16. [here](https://github.com/collinbarrett/wp-vps-build-guide)) without any other configuration or tuning. Digging into if and why it would default to my /run and how I can change it...
